mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -0,0 +1,291 @@
+
+    long startTime = System.currentTimeMillis();
+    LOG.info("Committing job has started for table: {}, using location: {}", table,
+        generateJobLocation(conf, jobContext.getJobID()));
+
+    FileIO io = HiveIcebergStorageHandler.io(jobContext.getJobConf());
+    List<DataFile> dataFiles = dataFiles(jobContext, io, true);
+
+    if (dataFiles.size() > 0) {
+      // Appending data files to the table
+      AppendFiles append = table.newAppend();
+      dataFiles.forEach(append::appendFile);
+      append.commit();
+      LOG.info("Commit took {} ms for table: {} with {} file(s)", System.currentTimeMillis() - startTime, table,
+          dataFiles.size());
+      LOG.debug("Added files {}", dataFiles);
+    } else {
+      LOG.info("Commit took {} ms for table: {} with no new files", System.currentTimeMillis() - startTime, table);
+    }
+
+    cleanup(jobContext);
+  }
+
+  /**
+   * Removes the generated data files, if there is a commit file already generated for them.
+   * The cleanup at the end removes the temporary directory as well.
+   * @param jobContext The job context
+   * @param status The status of the job
+   * @throws IOException if there is a failure deleting the files
+   */
+  @Override
+  public void abortJob(JobContext jobContext, int status) throws IOException {
+    String location = generateJobLocation(jobContext.getJobConf(), jobContext.getJobID());
+    LOG.info("Job {} is aborted. Cleaning job location {}", jobContext.getJobID(), location);
+
+    FileIO io = HiveIcebergStorageHandler.io(jobContext.getJobConf());
+    List<DataFile> dataFiles = dataFiles(jobContext, io, false);
+
+    // Check if we have files already committed and remove data files if there are any
+    if (dataFiles.size() > 0) {
+      Tasks.foreach(dataFiles)
+          .retry(3)
+          .suppressFailureWhenFinished()
+          .onFailure((file, exc) -> LOG.debug("Failed on to remove data file {} on abort job", file.path(), exc))
+          .run(file -> io.deleteFile(file.path().toString()));
+    }
+
+    cleanup(jobContext);
+  }
+
+  /**
+   * Cleans up the jobs temporary location.
+   * @param jobContext The job context
+   * @throws IOException if there is a failure deleting the files
+   */
+  private void cleanup(JobContext jobContext) throws IOException {
+    String location = generateJobLocation(jobContext.getJobConf(), jobContext.getJobID());
+    LOG.info("Cleaning for job: {} on location: {}", jobContext.getJobID(), location);
+
+    // Remove the job's temp directory recursively.
+    // Intentionally used foreach on a single item. Using the Tasks API here only for the retry capability.
+    Tasks.foreach(location)
+        .retry(3)
+        .suppressFailureWhenFinished()
+        .onFailure((file, exc) -> LOG.debug("Failed on to remove directory {} on cleanup job", file, exc))
+        .run(file -> {
+          Path toDelete = new Path(file);
+          FileSystem fs = Util.getFs(toDelete, jobContext.getJobConf());
```

Comment:
Nit: I would prefer if cleanup happened by removing the expected task commit files one-by-one rather than deleting a directory because it could use `FileIO`. I understand that this is intended to drop the folder as well for stores that track folders. Maybe a follow-up to add a `deletePrefix` to `FileIO` would fix it.